osu.Game.Tests/Chat/TestSceneChannelManager.cs

```diff
@@ -0,0 +1,104 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Linq;
+using NUnit.Framework;
+using osu.Framework.Allocation;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Testing;
+using osu.Game.Online.API;
+using osu.Game.Online.API.Requests;
+using osu.Game.Online.Chat;
+using osu.Game.Tests.Visual;
+using osu.Game.Users;
+
+namespace osu.Game.Tests.Chat
+{
+    [HeadlessTest]
+    public class TestSceneChannelManager : OsuTestScene
+    {
+        private ChannelManager channelManager;
+        private int currentMessageId;
+
+        [SetUp]
+        public void Setup() => Schedule(() =>
+        {
+            var container = new ChannelManagerContainer();
+            Child = container;
+            channelManager = container.ChannelManager;
+        });
+
+        [SetUpSteps]
+        public void SetUpSteps()
+        {
+            AddStep("register request handling", () =>
+            {
+                currentMessageId = 0;
+
+                ((DummyAPIAccess)API).HandleRequest = req =>
+                {
+                    switch (req)
+                    {
+                        case JoinChannelRequest _:
+                            return true;
```

Comment:
I thought this will keep the request pending indefinitely and not set its `Joined` state, but turns out it sets it before even queueing the request and leaves from the channel if the request fails:
https://github.com/ppy/osu/blob/9132c42f875576a4d7c9e74cccb1ecc1a9592826/osu.Game/Online/Chat/ChannelManager.cs#L379-L410

Maybe still worth calling `TriggerSuccess` anyways yeah?

Reply:
Sure. This was copied from `TestSceneChatOverlay`.

Reply:
@smoogipoo are you planning on making a change here?